### PR TITLE
Replace takeBoolPointer, takeIntPointer, takeInt64Pointer with generic addrOf

### PR DESCRIPTION
**Pull request by Didainius**
Before Go had generics, we used type based functions to get addresses of variables.

These were functions in `govcd/api.go`:
* `takeBoolPointer`
* `takeIntPointer`
* `takeInt64Pointer`

The above functions are now replaced by generic function `func addrOf[T any](variable T) *T`. 

The content of this PR is "search and replace". No user-facing changes.

### CODE REVIEW

File: vcd/resource_vcd_edgegateway.go [hunk 1]
```diff
@@ -307,7 +307,7 @@ func resourceVcdEdgeGatewayCreate(d *schema.ResourceData, meta interface{}) erro
 	if fipsModeEnabled, ok := d.GetOkExists("fips_mode_enabled"); ok {
 		fipsModeEnabledBool := fipsModeEnabled.(bool)
 		log.Printf("[TRACE] edge gateway creation. FIPS mode was set with value %t", fipsModeEnabledBool)
-		egwConfiguration.Configuration.FipsModeEnabled = takeBoolPointer(fipsModeEnabledBool)
+		egwConfiguration.Configuration.FipsModeEnabled = addrOf(fipsModeEnabledBool)
```
Comment (dataclouder) [inline vcd/resource_vcd_edgegateway.go:310]:
```suggestion
		egwConfiguration.Configuration.FipsModeEnabled = &fipsModeEnabledBool
```

File: vcd/resource_vcd_global_role.go [hunk 2]
```diff
@@ -77,7 +77,7 @@ func resourceGlobalRoleCreate(ctx context.Context, d *schema.ResourceData, meta
 		Name:        globalRoleName,
 		Description: d.Get("description").(string),
 		BundleKey:   types.VcloudUndefinedKey,
-		PublishAll:  takeBoolPointer(publishToAllTenants),
+		PublishAll:  addrOf(publishToAllTenants),
```
Comment (dataclouder) [inline vcd/resource_vcd_global_role.go:80]:
```suggestion
		PublishAll:  &publishToAllTenants,
```

File: vcd/resource_vcd_global_role.go [hunk 3]
```diff
@@ -209,7 +209,7 @@ func resourceGlobalRoleUpdate(ctx context.Context, d *schema.ResourceData, meta
 	if d.HasChange("name") || d.HasChange("description") || d.HasChange("publish_to_all_tenants") {
 		globalRole.GlobalRole.Name = globalRoleName
 		globalRole.GlobalRole.Description = d.Get("description").(string)
-		globalRole.GlobalRole.PublishAll = takeBoolPointer(publishToAllTenants)
+		globalRole.GlobalRole.PublishAll = addrOf(publishToAllTenants)
```
Comment (dataclouder) [inline vcd/resource_vcd_global_role.go:212]:
```suggestion
		globalRole.GlobalRole.PublishAll = &publishToAllTenants
```

File: vcd/resource_vcd_nsxt_alb_pool.go [hunk 4]
```diff
@@ -425,7 +425,7 @@ func getNsxtAlbPoolType(d *schema.ResourceData) (*types.NsxtAlbPool, error) {
 
 	caCertificateRefs, commonNameCheckEnabled, domainNames := getCertificateTypes(d)
 	albPoolConfig.CaCertificateRefs = caCertificateRefs
-	albPoolConfig.CommonNameCheckEnabled = takeBoolPointer(commonNameCheckEnabled)
+	albPoolConfig.CommonNameCheckEnabled = addrOf(commonNameCheckEnabled)
```
Comment (dataclouder) [inline vcd/resource_vcd_nsxt_alb_pool.go:428]:
```suggestion
	albPoolConfig.CommonNameCheckEnabled = &commonNameCheckEnabled
```

File: vcd/resource_vcd_rights_bundle.go [hunk 5]
```diff
@@ -75,7 +75,7 @@ func resourceRightsBundleCreate(ctx context.Context, d *schema.ResourceData, met
 		Name:        rightsBundleName,
 		Description: d.Get("description").(string),
 		BundleKey:   types.VcloudUndefinedKey,
-		PublishAll:  takeBoolPointer(publishToAllTenants),
+		PublishAll:  addrOf(publishToAllTenants),
```
Comment (dataclouder) [inline vcd/resource_vcd_rights_bundle.go:78]:
```suggestion
		PublishAll:  &publishToAllTenants,
```

File: vcd/resource_vcd_rights_bundle.go [hunk 6]
```diff
@@ -205,7 +205,7 @@ func resourceRightsBundleUpdate(ctx context.Context, d *schema.ResourceData, met
 	if d.HasChange("name") || d.HasChange("description") || d.HasChange("publish_to_all_tenants") {
 		rightsBundle.RightsBundle.Name = rightsBundleName
 		rightsBundle.RightsBundle.Description = d.Get("description").(string)
-		rightsBundle.RightsBundle.PublishAll = takeBoolPointer(publishToAllTenants)
+		rightsBundle.RightsBundle.PublishAll = addrOf(publishToAllTenants)
```
Comment (dataclouder) [inline vcd/resource_vcd_rights_bundle.go:208]:
```suggestion
		rightsBundle.RightsBundle.PublishAll = &publishToAllTenants
```

File: vcd/resource_vcd_vapp_vm_tools.go [hunk 7]
```diff
@@ -995,17 +995,17 @@ func updateCustomizationSection(customizationInterface interface{}, d *schema.Re
 				// AdminAutoLogonEnabled=false if number_of_auto_logons == 0
 				// AdminAutoLogonEnabled=true if number_of_auto_logons > 0
 				isMoreThanZero := nrTimesForLogin.(int) > 0
-				customizationSection.AdminAutoLogonEnabled = takeBoolPointer(isMoreThanZero)
+				customizationSection.AdminAutoLogonEnabled = addrOf(isMoreThanZero)
```
Comment (dataclouder) [inline vcd/resource_vcd_vapp_vm_tools.go:998]:
```suggestion
				customizationSection.AdminAutoLogonEnabled = &isMoreThanZero
```

File: vcd/resource_vcd_vm_affinity_rule.go [hunk 8]
```diff
@@ -130,8 +131,8 @@ func resourceToAffinityRule(d *schema.ResourceData, meta interface{}) (*types.Vm
 
 	var vmAffinityRuleDef = &types.VmAffinityRule{
 		Name:        name,
-		IsEnabled:   takeBoolPointer(enabled),
-		IsMandatory: takeBoolPointer(required),
+		IsEnabled:   addrOf(enabled),
+		IsMandatory: addrOf(required),
```
Comment (dataclouder) [inline vcd/resource_vcd_vm_affinity_rule.go:135]:
```suggestion
		IsEnabled:   &enabled,
		IsMandatory: &required,
```

**Review by dataclouder — APPROVED** [review]
A few replacements are needed, where we use addrOf on a local variable

**Review by adambarreiro — APPROVED** [review]
Thanks!

**Comment by Didainius** [comment]
> Just putting out one naming idea: `addrOf` could also be `pointerOf`.

I think we had this misunderstanding previously (I might be the one who contributed to that), but the point is that `addrOf` is closer IMO. 

* Notation `&variable` is literally "address of `variable`"
* "pointer of/to `variable`" would be `*variable`


We are using `addrOf` in the places where we would like to input "address of literal value" (e.g. `&true`, `&4`, etc), but this is not possible in Go so here comes this function `addrOf` which makes it possible to do inline:
```go
addrOf(true)
addrOf(4)
```
If we hadn't one, we'd have to do an intermediate step for declaring the variable and taking the address of it
```go
boolVar := true
&boolVar

intVar := 4
&intVar
```

**Review by adezxc — APPROVED** [review]

**Pull request merged**